### PR TITLE
Update libraries

### PR DESCRIPTION
**Pull request by cmonfortep**
Task/Issue URL: https://app.asana.com/0/414730916066338/1200541903185345/f
Tech Design URL: 
CC: 

**Description**:
This PR updates the following libraries:

- version.androidx.room: 2.2.6 -> 2.3.0
- version.kotlin=1.4.10 -> 1.5.20
- version.androidx.appcompat=1.2.0 -> 1.3.0
- version.androidx.webkit=1.3.0 -> 1.4.0
- version.androidx.core=1.3.2 → 1.5.0
- version.androidx.fragment=1.2.5 → 1.3.5
- version.androidx.work=2.4.0 -> 2.5.0
- version.okhttp3=4.9.0 -> 4.9.1
- version.mockito=3.4.6 ->3.10.0

**Steps to test this PR**:
1. App should build and pass all the tests.
1. Install the app and perform any action to ensure no unexpected issues.

Smoke test:
- Onboarding CTAs
- Add bookmarks and favorites
- disable/enable privacy protection
- share feedback
- fireproof a site and clear all data
- opening multiple tabs (and closing tabs + undo)

Note: All the changes have been tested already on API 21

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: app/src/androidTest/java/com/duckduckgo/app/browser/downloader/BlobConverterInjectorJsTest.kt
Comment (cmonfortep) [inline app/src/androidTest/java/com/duckduckgo/app/browser/downloader/BlobConverterInjectorJsTest.kt:41]: Currently, `spy` is not supported on API < 24
https://github.com/mockito/mockito/issues/2075#issuecomment-723507001

**Review by aitorvs — APPROVED** [review]
Thanks for doing this @cmonfortep

**Comment by cmonfortep** [comment]
Thanks @aitorvs. I'm blocking this PR until we merge and release https://github.com/duckduckgo/Android/pull/1274, just to avoid any unexpected issue during the rollout.

File: versions.properties
Comment (aitorvs) [inline versions.properties:37]: I am not sure this one is used. If it is, maybe we should remove the one we have in the main `build.gradle` and used this one. They are easy to get out of sync, in fact, they had different versions :/

File: build.gradle
Comment (cmonfortep) [inline build.gradle:17]: @aitorvs I think this is the way we can avoid having to duplicate the kotlin version in 2 files, but not 100% sure. 

By executing `./gradlew buildEnvironment` seems this will work: `org.jetbrains.kotlin:kotlin-gradle-plugin:{require 1.5.20; reject _} -> 1.5.20`, so I guess is ok

Comment (aitorvs) [reply build.gradle:17]: That's interesting! I seems it works because the `_` is not really a valid version and then it falls back to any other (latest) version we might have. I think a more secure approach might be the following, in the root `build.gradle`

```groovy
buildscript {
    def versionPropsFile = file('versions.properties')
    Properties versionProps = new Properties()

    if (versionPropsFile.canRead()) {
        versionProps.load(new FileInputStream(versionPropsFile))
    }

    ext.kotlin_version = versionProps['version.kotlin']
    ...
```

**Pull request merged**
